### ISSUE TITLE
SC2034 triggered when assignment is used only in a trap

### ISSUE THREAD

**Issue by cdown** [koalaman/shellcheck#1095]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2034
- My shellcheck version (`shellcheck --version` or "online"): 0.4.7
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash -e

trap '(( foo )) && echo foo' EXIT
foo=1
```

#### Here's what shellcheck currently says:

```
In trap_use line 4:
foo=1
^-- SC2034: foo appears unused. Verify it or export it.
```

#### Here's what I wanted or expected to see:

Nothing, this is used in the trap.

**Comment by nihilus** [comment]
That is some bad coding, use a function instead and assign that to the trap, at least for readability.

**Comment by cdown** [comment]
@nihilus In this case it literally makes no difference. An inline trap for something tiny is hardly invalid.

**Comment by nihilus** [comment]
Never said it was invalid, just bad practice.

**Comment by koalaman** [comment]
You're right of course.

This was a design decision. Any code in a string literal will be treated as opaque by ShellCheck for reasons mentioned in #1038. 

(But, confusingly and contrary to this, it will still look through trap strings for the substring `$foo` )
